pigeon/notifications.dart

```diff
@@ -8,6 +8,33 @@ import 'package:pigeon/pigeon.dart';
   kotlinOptions: KotlinOptions(package: 'com.zulip.flutter'),
 ))
 
+/// Corresponds to `androidx.core.app.NotificationChannelCompat`
+///
+/// See:
+///   https://developer.android.com/reference/androidx/core/app/NotificationChannelCompat
+///   https://developer.android.com/reference/androidx/core/app/NotificationChannelCompat.Builder#Builder(java.lang.String,int)
+class NotificationChannel {
+  NotificationChannel({
```

Comment:
Ah, just to clarify what I meant at https://github.com/zulip/zulip-flutter/pull/795#discussion_r1684946174 because it might have been confusing:

I think having a doc comment on this Pigeon/Dart constructor is fine and potentially helpful. What I meant was that the doc on this constructor should link to the doc of the upstream builder class as a whole, and not to the specific point on that page for the builder's constructor.

So like this:

```suggestion
/// See:
///   https://developer.android.com/reference/androidx/core/app/NotificationChannelCompat
///   https://developer.android.com/reference/androidx/core/app/NotificationChannelCompat.Builder
class NotificationChannel {
  NotificationChannel({
```

but also that second link would be fine to move back to being a doc comment on this NotificationChannel constructor rather than on the NotificationChannel class.